lib/MusicBrainz/Server/Data/Tag.pm

```diff
@@ -96,6 +96,55 @@ sub get_cloud
     return $data;
 }
 
+sub rename_for_user {
```

Comment:
The problem with this is that it doesn't update the aggregate tag tables, only the raw ones. So the new tags will have broken counts and not display in the editing UI at all in some cases.

Ideally we'd handle this via triggers, but that's not the current situation unfortunately. This should really only be using `Data::EntityTag` methods to perform updates (which is accessible via `->tags` on each entity model), but those only operate on one entity and one tag at present. They'd have to be modified to operate on a list of entities.

Reply:
So basically use upvote/downvote/withdraw?

Reply:
Yeah, except for the last bit I mentioned there ("those only operate on one entity and one tag at present. They'd have to be modified to operate on a list of entities").